autopush/log_check.py

```diff
@@ -0,0 +1,49 @@
+import uuid
+
+import cyclone.web
+from twisted.internet.threads import deferToThread
+
+from autopush.endpoint import AutoendpointHandler
+
+
+class LogCheckError(Exception):
+    pass
+
+
+class LogCheckHandler(AutoendpointHandler):
+
+    def initialize(self, ap_settings):
+        self.ap_settings = ap_settings
+        self.request_id = str(uuid.uuid4())
+        self._client_info = self._init_info()
+
+    def _throw(self):
+        raise LogCheckError("LogCheck")
+
+    def _response_err(self, failure):
+        self.log.failure(format="LogCheck", failure=failure,
+                         status_code=418, errno=999, **self._client_info)
+        self._write_response(418, 999, message="FAILURE:Success",
+                             reason="Test Failure")
+
+    @cyclone.web.asynchronous
+    def get(self, errType=None):
+        """HTTP GET
+
+        Generate a dummy error message for logging
+
+        """
+        if not errType:
+            errType = "error"
+        else:
+            errType = errType.lower()
+        if 'error' in errType:
+            self.log.error(format="Test Error Message",
+                           status_code=418, errno=0,
+                           **self._client_info)
+            self._write_response(418, 999, message="ERROR:Success",
+                                 reason="Test Error")
+        if 'crit' in errType:
```

Comment:
Why not just...
```python
if 'crit' in errType:
    try:
        raise LogCheckError("Logcheck")
    except:
        self.log.failure("Test error message", ...)
```

log.failure should automatically pull out the active exception if no failure is passed to it (the docs seem to indicate that at least)